tests/failpoints/cases/test_snap.rs

```diff
@@ -436,3 +436,33 @@ fn test_receive_old_snapshot() {
 
     fail::remove(peer_2_handle_snap_mgr_gc_fp);
 }
+
+#[test]
+fn test_snapshot_after_isolation() {
```

Comment:
Should add comment explaining the purpose of the case.

Reply:
It can pass successfully without the patch.

Reply:
It can't pass with the `before_no_ready_gen_snap_task` fail point enabled, IMO the `before_no_ready_gen_snap_task` fail point is a simple way to simulate a non-empty `Ready` that without any `committed_entries`

Reply:
Comment has been added.
This test cannot be passed without the patch on my laptop. I guess you forget to add the failpoint.